lib/SymbolGraphGen/Symbol.cpp

```diff
@@ -709,9 +767,30 @@ bool Symbol::supportsKind(DeclKind Kind) {
   case DeclKind::Var: LLVM_FALLTHROUGH;
   case DeclKind::Subscript: LLVM_FALLTHROUGH;
   case DeclKind::TypeAlias: LLVM_FALLTHROUGH;
-  case DeclKind::AssociatedType:
+  case DeclKind::AssociatedType: LLVM_FALLTHROUGH;
+  case DeclKind::Extension:
     return true;
   default:
     return false;
   }
 }
+
+AccessLevel Symbol::getEffectiveAccessLevel(const ExtensionDecl *ED) {
```

Comment:
Shouldn't just `ED.getFormalAccess()` compute this? There is no way this computation doesn't exist anywhere else in the compiler. We shouldn't reinvent the wheel here.

Reply:
No, unfortunately it doesn't. I think `ED.getMaxAccessLevel()` computes the maximum access level one of its members **can** have based on the extended type and types used in constraints.

I want the effective access level for this ExtensionDecl to be `internal`.
```swift
extension String {
    func foo() { }
}
```

I spent some time searching for a fitting function, but couldn't find one. It kinda makes sense as this logic isn't really relevant to the rest of the compiler, as the extension block itself isn't an accessible symbol in Swift code.

Reply:
I extended the comment in Symbol.h accordingly in fc872b09428fc858852befb64bc4b10680a66d99.